Student Name: Marion Osong
Student Pace: DSF-FT12-Hybrid
Instructor Name: Samuel Karu

## **MOVIE COMPANY PROJECT.** ##

### *Box office perfomance analysis for new Movie studio.* ###

Student Name: Marion Osong (Group 2)

Student Pace: DSF-FT12-Hybrid

Instructor Name: Samuel Karu

## 🎬Introduction ##


## 🎯Problem statement ##
"Our company aims to establish a new movie studio but lacks experience in film production. To guide the development of a successful film portfolio, we need to analyze historical box office data to identify patterns and factors that contribute to a film's financial success. This analysis will provide actionable insights to inform strategic decisions on genre selection, casting, budgeting, and marketing, ensuring the creation of films with high potential for profitability."
## Objectives ##
### 1. Undestanding and analyze historical Box Office Data.* ####
The Boxoffice Company is a leading global provider of media, technology, and data solutions for the film industry.They partner with top studios, search platforms, and over 300 exhibition partners to connect with moviegoers through tools for ticketing, marketing, and business intelligence.
- **Goal:** 
To identify patterns that inform future project selections.
To also understand the genres that have historically performed well.
      
### *2. Understanding the dataset.* ###
- **Tasks**
   - Understand the technical terms used in the dataset.
   - Review the shape and type of the data, understand the numerical and categorical columns. 
   - Handle missing values (drop or replace). 
   - Identify and remove duplicates. 
   - Drop unnecessary columns and add new ones as needed<p>

3. Business objectives.
- **Tasks**
- Analyze cost structures of successful films to identify areas for cost optimization.

- Develop models to predict the return on investment (ROI) for different types of films.

### *4. statistics.* ###
- **Trend Analysis**
-

### **DATA LOADING.** ###

This step is crucial for ensuring that data is accessible, consistent, and ready for subsequent analysis or modeling

In [33]:
# import the necessaty libraries for data analysis and visualization.

import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import sqlite3
import seaborn as sns
%matplotlib inline



In [34]:
# load the dataset and display the first 5 rows.
df = pd.read_csv('../zipped-data/bom.movie_gross.csv/bom.movie_gross.csv')
df.head()


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [35]:
# understanding the dataframe and its structure. (Get column attributes and data types.)

df.info() 


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [36]:
df.columns

Index(['title', 'studio', 'domestic_gross', 'foreign_gross', 'year'], dtype='object')

In [37]:
df.shape

(3387, 5)

In [38]:
df.describe(include='all')

title studio  domestic_gross foreign_gross         year
count        3387   3382    3.359000e+03          2037  3387.000000
unique       3386    257             NaN          1204          NaN
top     Bluebeard    IFC             NaN       1200000          NaN
freq            2    166             NaN            23          NaN
mean          NaN    NaN    2.874585e+07           NaN  2013.958075
std           NaN    NaN    6.698250e+07           NaN     2.478141
min           NaN    NaN    1.000000e+02           NaN  2010.000000
25%           NaN    NaN    1.200000e+05           NaN  2012.000000
50%           NaN    NaN    1.400000e+06           NaN  2014.000000
75%           NaN    NaN    2.790000e+07           NaN  2016.000000
max           NaN    NaN    9.367000e+08           NaN  2018.000000

In [39]:
# Total number of null values in the dataset.
df.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [40]:
# Total number of duplicates in the dataset.
df.duplicated().sum()

0

In [41]:
# Total number of the categorical data in the dataset.
df.describe(include='object')

title studio foreign_gross
count        3387   3382          2037
unique       3386    257          1204
top     Bluebeard    IFC       1200000
freq            2    166            23

In [42]:
# Total number of numerical data in the dataset.
df.describe(include='number')

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

### *Summary of the dataset.* ###

- Has a total number of 5 colums and 3387 rows in total.
- 

### **DATA CLEANING.** ###

### *standadizing the dataset.* ####

In [43]:
# checking for duplicates in the dataset.
df.duplicated().sum()

0

In [44]:
# handling the null values in the numerical data.

null_values = df.isnull().sum()

#Displaying only columns with null values.
null_values = null_values[null_values > 0]

#Checking percentage of null values.
null_values_percentage = (null_values/len(df)) * 100

#Display null values with percentage.
null_data = pd.DataFrame({'Null Value': null_values, 'Percentage': null_values_percentage})
null_data

Null Value  Percentage
studio                   5    0.147623
domestic_gross          28    0.826690
foreign_gross         1350   39.858282

### *Dropping rows and columns with null values.* ###

In [45]:
# Drop columns with more than 60% missing values.
df = df.dropna(thresh=len(df) * 0.4, axis=1)
